src/app/modules/main/chat_section/controller.nim

```diff
@@ -162,18 +166,33 @@ method setActiveItemSubItem*(self: Controller, itemId: string, subItemId: string
 
   self.delegate.activeItemSubItemSet(self.activeItemId, self.activeSubItemId)
 
-method removeActiveFromThisChat*(self: Controller, itemId: string) =
-  if self.activeItemId != itemId:
-    return
+method deleteChannel*(self: Controller, itemId: string) =
+  let allChats = self.communityService.getChats(self.getMySectionId())
+  self.activeSubItemId = ""
+  if allChats.len == 0:
+    self.activeItemId = ""
+  else:
+    self.activeItemId = self.getMySectionId() & allChats[0].id
+
+  self.communityService.deleteCommunityChat(self.getMySectionId(), itemId)
 
+method deleteChat*(self: Controller, itemId: string) =
   let allChats = self.chatService.getAllChats()
-
   self.activeSubItemId = ""
   if allChats.len == 0:
     self.activeItemId = ""
   else:
     self.activeItemId = allChats[0].id
 
+  echo "Todo for chat "
+
+method removeChat*(self: Controller, itemId: string) =
+  if self.isCommunitySection:
+    self.deleteChannel(itemId)
+  else:
+    self.deleteChat(itemId)
+
+method onCommunityChannelDeleted*(self: Controller) =
```

Comment:
@Khushboo-dev-cpp this is unnecessary here, cause you already have all the data you need when channel is deleted and you're in `onCommunityChannelDeleted` method. So please remove this one from here and put a logic in `module.nim` something like this:
```
method onCommunityChannelDeleted*(self: Module, communityId: string, chatId: string) =
  if(not self.chatContentModules.contains(chatId)):
    return

  self.view.chatsModel().removeItemById(chatId)
  self.removeSubmodule(chatId)
  
  if(self.view.chatsModel().getCount() == 0):
    return

  # set the first channel as the active one in model
  let item = self.view.chatsModel().getItemAtIndex(0) # not sure if we have this method, if not add it
  if(item.subItems.getCount() == 0):
    self.setActiveItemSubItem(item.id, "")
  else
    let subItem = item.subItems.getItemAtIndex(0) # not sure if we have this method, if not add it
    self.setActiveItemSubItem(item.id, subItem.id)
```